libcloud/common/aws.py

```diff
@@ -178,6 +180,97 @@ def _get_aws_auth_param(self, params, secret_key, path='/'):
         return b64_hmac.decode('utf-8')
 
 
+class V4SignedAWSConnection(AWSTokenConnection):
+
+    def add_default_params(self, params):
+        params['Version'] = self.version
+        return params
+
+    def pre_connect_hook(self, params, headers):
+        now = datetime.utcnow()
```

Comment:
But as this currently works, I deduce that AWS is not that strict about this header :wink: